src/ts/component/block/dataview.tsx

```diff
@@ -705,14 +703,12 @@ const BlockDataview = observer(class BlockDataview extends React.Component<Props
 			});
 		};
 
-		UtilObject.checkDefaultTemplate(details.type, templateId, (res) => {
-			if (!hasSources || !res) {
-				template = null;
-				C.BlockDataviewViewUpdate(rootId, block.id, view.id, { ...view, defaultTemplateId: Constant.templateId.blank }, create);
-			} else {
-				create();
-			};
-		});
+		if (defaultTemplate.isArchived || defaultTemplate.isDeleted) {
+			template = null;
+			C.BlockDataviewViewUpdate(rootId, block.id, view.id, { ...view, defaultTemplateId: Constant.templateId.blank }, create);
```

Comment:
Remove logic for updating the view, archived or deleted template should not display as selected option.